core/input/input_map.cpp

```diff
@@ -106,7 +106,9 @@ List<Ref<InputEvent>>::Element *InputMap::_find_event(Action &p_action, const Re
 
 		int device = e->get_device();
 		if (device == ALL_DEVICES || device == p_event->get_device()) {
-			if (e->action_match(p_event, p_pressed, p_strength, p_raw_strength, p_action.deadzone)) {
+			if (p_exact_match && e->shortcut_match(p_event)) {
+				return E;
+			} else if (!p_exact_match && e->action_match(p_event, p_pressed, p_strength, p_raw_strength, p_action.deadzone)) {
 				return E;
```

Comment:
This is where the check actually happens.

"Action Match" has remained "fuzzy" as it was previously, and "Shortcut Match" is used to do the exact check.

This could potentially be changed to pass the "p_exact" parameter to action_match and just do the comparison inside that method.